### ISSUE TITLE
Enhance metadata to simulate partition pruning/min-max pruning of indexes

### ISSUE THREAD

**Issue by apoorvedave1** [microsoft/hyperspace#109]
### Describe the issue

Depends on #108 
Investigate options to enhance index metadata to support simulated partition pruning on the index files

### To Reproduce

na

### Expected behavior

na
### Environment 

na

**Comment by rapoth** [comment]
I am closing this in favor of #146 

@sezruby Please feel free to include this as a usecase in the optimizer side.
